### ISSUE TITLE
Bugs in JNA binding (boolean and null)

### ISSUE THREAD

**Issue by mandoway** [ElektraInitiative/libelektra#4368]
1. When setting a Key value to a boolean with the JNA binding, the actuallly stored value will be "true" or "false" and not "1" or "0" (https://www.libelektra.org/decisions/boolean).
https://github.com/ElektraInitiative/libelektra/blob/f1b341bbe3cf3ae75df4920282c80dd6ad1c679b/src/bindings/jna/libelektra/src/main/java/org/libelektra/Key.java#L223-L226

2. The value of a Key cannot be set to null once it has a value stored.

## Steps to Reproduce the Problem

1. Setting a boolean
```java
key.setBoolean(true);
```

2. Doesn't work or exist:
```java
key.setString(null);
or 
key.setNull();
```

## Expected Result

1. Boolean value should be "1" or "0"
2. Setting a value null should be possible

## Actual Result

1. Boolean value is actually "true" or "false"
2. Setting a value null is not possible

## System Information

- Elektra: master
- Operating System: Windows 10

**Comment by kodebach** [comment]
> 3\. Doesn't work or exist:
> 
> ```java
> key.setString(null);
> or 
> key.setNull();
> ```

In the current API only `key.setNull` would make sense, alternatively `key.setBinary(null)` would be an option. I'd prefer the explicit `setNull`, but `setBinary(null)` would be consistent with the underlying C API, where a null key is always binary.

**Comment by tucek** [comment]
I'm going to fix this tonight. I agree with @kodebach `Key#setNull()` is currently the consistent approach for the Java API.

**Comment by tucek** [comment]
@kodebach @markus2330 (ad https://www.libelektra.org/decisions/null)
Should we be able to determine whether a key actually has a value? How do i get this from the C API?

**Comment by tucek** [comment]
created draft PR https://github.com/ElektraInitiative/libelektra/pull/4372

**Comment by tucek** [comment]
ready to be merged with PR #4372

**Comment by markus2330** [comment]
Fixed via https://github.com/ElektraInitiative/libelektra/pull/4372

@mandoway @Gratla does this now need any adoptions for the Kotlin binding? Would be great to see this fixed for T3.

**Comment by mandoway** [comment]
> does this now need any adoptions for the Kotlin binding?

For booleans: no.
Since we use getters/setters from JNA.

For null values: yes.
We currently set the value to an empty string when `null` is passed to the `set()` function.

**Comment by mandoway** [comment]
Kotlin binding is adjusted to these fixes in PR #4374

**Comment by tucek** [comment]
@kodebach i think we can close this issue, since jna and kotlin fixes are merged?
